src/event_gate_lambda.py

```
@@ -80,17 +81,33 @@
-TOKEN_PROVIDER_URL = CONFIG["token_provider_url"]
-# Add timeout to avoid hanging requests; wrap in robust error handling so failures are explicit
+# Initialize token public keys
+TOKEN_PROVIDER_URL = CONFIG.get("token_provider_url")
+TOKEN_PUBLIC_KEYS_URL = CONFIG.get("token_public_keys_url") or CONFIG.get("token_public_key_url")
+
 try:
-    response_json = requests.get(CONFIG["token_public_key_url"], verify=False, timeout=5).json()  # nosec external
-    token_public_key_encoded = response_json["key"]
-    TOKEN_PUBLIC_KEY: Any = serialization.load_der_public_key(base64.b64decode(token_public_key_encoded))
-    logger.debug("Loaded TOKEN_PUBLIC_KEY")
+    response_json = requests.get(TOKEN_PUBLIC_KEYS_URL, verify=False, timeout=5).json()
```

Ruff (0.14.4)

Probable use of `requests` call with `verify=False` disabling SSL certificate checks

(S501)